src/components/ModalWithForm.js

```diff
@@ -6,6 +6,10 @@ export default class ModalWithForm extends Modal {
     this._modalForm = this._modalElement.querySelector(".modal__form");
     this._handleFormSubmit = handleFormSubmit;
   }
+  open() {
+    this._modalForm.querySelector(".modal__button").innerText = "Save";
+    super.open();
+  }
   _getInputValues() {
     const _modalInputs = [...this._modalForm.querySelectorAll(".modal__input")];
```

Comment:
**Could be improved**

All elements that are found every time you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `generateCard`) only **1 time** not to waste resources on searching them again and again.

`Class fields` (the so called **class variables** with `this`) are accessible inside any method of the class. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form` , `this._likeButton` and so on.